homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,284 @@
+"""Module contains the CompitClimate class for controlling climate entities."""
+
+from enum import Enum
+import logging
+from typing import Any
+
+from compit_inext_api import Device, Param, Parameter
+from propcache.api import cached_property
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER_NAME
+from .coordinator import CompitConfigEntry, CompitDataUpdateCoordinator
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+ATTR_TEMPERATURE = "temperature"
+PARAM_PRESET_MODE = "__trybpracytermostatu"
+PARAM_FAN_MODE = "__trybaero"
+PARAM_HVAC_MODE = "__trybpracyinstalacji"
+PARAM_CURRENT_TEMPERATURE = "__tpokojowa"
+PARAM_TARGET_TEMPERATURE = "__tpokzadana"
+PARAM_SET_TARGET_TEMPERATURE = "__tempzadpracareczna"
+CLIMATE_DEVICE_CLASS = 10
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+class CompitHVACMode(Enum):
+    """Enum for available HVAC modes."""
+
+    HEAT = 0
+    OFF = 1
+    COOL = 2
+
+
+COMPIT_MODE_MAP = {
+    CompitHVACMode.COOL: HVACMode.COOL,
+    CompitHVACMode.HEAT: HVACMode.HEAT,
+    CompitHVACMode.OFF: HVACMode.OFF,
+}
+
+HVAC_MODE_TO_COMPIT_MODE = {v: k for k, v in COMPIT_MODE_MAP.items()}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: CompitConfigEntry,
+    async_add_devices: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the CompitClimate platform from a config entry."""
+
+    coordinator: CompitDataUpdateCoordinator = entry.runtime_data
+    async_add_devices(
+        [
+            CompitClimate(
+                coordinator,
+                device,
+                device_definition.parameters,
+                device_definition.name,
+            )
+            for gates in coordinator.gates
+            for device in gates.devices
+            if (
+                device_definition := next(
+                    (
+                        definition
+                        for definition in coordinator.device_definitions.devices
+                        if definition.code == device.type
+                    ),
+                    None,
+                )
+            )
+            is not None
+            if (device_definition.device_class == CLIMATE_DEVICE_CLASS)
```

Comment:
Maybe it doesn't make sense, but would it make sense if this was a dict with the code as the key? This way it could just be a dict lookup instead of iterating over all the definitions